src/VisualStudio/Core/Def/Implementation/ProjectSystem/RunningDocumentTableEventTracker.cs

```diff
@@ -0,0 +1,258 @@
+        {
+            if (dwReadLocksRemaining + dwEditLocksRemaining == 0)
+            {
+                _foregroundAffinitization.AssertIsForeground();
+                if (_runningDocumentTable.IsDocumentInitialized(docCookie))
+                {
+                    _listener.OnCloseDocument(_runningDocumentTable.GetDocumentMoniker(docCookie));
+                }
+            }
+
+            return VSConstants.S_OK;
+        }
+
+        public int OnAfterSave(uint docCookie)
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+
+        public int OnAfterAttributeChange(uint docCookie, uint grfAttribs)
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+
+        public int OnAfterAttributeChangeEx(uint docCookie, uint grfAttribs, IVsHierarchy pHierOld, uint itemidOld, string pszMkDocumentOld, IVsHierarchy pHierNew, uint itemidNew, string pszMkDocumentNew)
+        {
+            // Did we rename?
+            if ((grfAttribs & (uint)__VSRDTATTRIB.RDTA_MkDocument) != 0)
+            {
+                _foregroundAffinitization.AssertIsForeground();
+                if (_runningDocumentTable.IsDocumentInitialized(docCookie) && TryGetBuffer(docCookie, out var buffer))
+                {
+                    _listener.OnRenameDocument(newMoniker: pszMkDocumentNew, oldMoniker: pszMkDocumentOld, textBuffer: buffer);
+                }
+            }
+
+            // Doc data reloaded is the most reliable way to know when a document has been loaded and may have a text buffer we can get.
+            if ((grfAttribs & (uint)__VSRDTATTRIB.RDTA_DocDataReloaded) != 0)
+            {
+                _foregroundAffinitization.AssertIsForeground();
+                if (_runningDocumentTable.IsDocumentInitialized(docCookie) && TryGetMoniker(docCookie, out var moniker) && TryGetBuffer(docCookie, out var buffer))
+                {
+                    _runningDocumentTable.GetDocumentHierarchyItem(docCookie, out var hierarchy, out _);
+                    _listener.OnOpenDocument(moniker, buffer, hierarchy);
+                }
+            }
+
+            if ((grfAttribs & (uint)__VSRDTATTRIB.RDTA_Hierarchy) != 0)
+            {
+                _foregroundAffinitization.AssertIsForeground();
+                if (_runningDocumentTable.IsDocumentInitialized(docCookie) && TryGetMoniker(docCookie, out var moniker))
+                {
+                    _runningDocumentTable.GetDocumentHierarchyItem(docCookie, out var hierarchy, out _);
+                    _listener.OnRefreshDocumentContext(moniker, hierarchy);
+                }
+            }
+
+            return VSConstants.S_OK;
+        }
+
+        public int OnBeforeDocumentWindowShow(uint docCookie, int fFirstShow, IVsWindowFrame pFrame)
+        {
+            // Doc data reloaded is not triggered for the underlying aspx.cs file when changes are made to the aspx file, so catch it here.
```

Comment:
The fix is implementing this method - when the generated method is added to the aspx.cs file, it does not trigger a DocDataReloaded attribute change.  However, OnBeforeDocumentWindowShow will get triggered.